src/theme/components/image.ts

```diff
@@ -1,17 +1,32 @@
 interface CustomImageBaseStyle {
-  size: keyof typeof imageSizes;
+  ratio: keyof typeof imageRatios;
+  size: keyof typeof imageSizes | keyof typeof imageSizesBasedOnHeight;
+  sizeBasedOn: "height" | "width";
 }
 interface CustomImageWrapperBaseStyle {
   ratio: keyof typeof imageRatios;
-  size: keyof typeof imageSizes;
+  size: keyof typeof imageSizes | keyof typeof imageSizesBasedOnHeight;
+  sizeBasedOn: "height" | "width";
 }
 
-const imageWrap = {
-  position: "relative",
-  width: "100%",
-  height: "auto",
-  overflow: "hidden",
+const sizes = {
+  xxxsmall: 32,
```

Comment:
Do you not need the specific "px" text in the value? Don't think you need to change anything but I wonder if the number 64 is the same as the string "64px" in css.

Reply:
Ah, I now see how you're using this. Never mind the comment!